src/Symfony/Bridge/Doctrine/Middleware/AutoClose/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware\AutoClose\Listener;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    /**
+     * @param \ArrayObject<string, int> $connectionExpiries
+     */
+    public function __construct(
+        private readonly \ArrayObject $connectionExpiries,
+        private ContainerInterface $container,
+    ) {
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $connectionExpiries = $this->connectionExpiries->getIterator();
+
+        $timestamp = time();
+
+        foreach ($connectionExpiries as $name => $expiry) {
+            if ($timestamp >= $expiry) {
+                $connection = $this->container->get(sprintf('doctrine.dbal.%s_connection', $name));
```

Comment:
```suggestion
                $connection = $this->container->get("doctrine.dbal.{$name}_connection");
```

Feel free to discard this micro-optimization if you want :)

Reply:
nah, I'm good with these optimisations! :)